pkgs/stdenv/cross/default.nix

```diff
@@ -41,25 +41,43 @@ in lib.init bootStages ++ [
       if crossSystem.isStatic
       then buildPackages.stdenvAdapters.makeStatic
       else lib.id;
+    stdenvNoCC = adaptStdenv (buildPackages.stdenv.override (old: rec {
+      buildPlatform = localSystem;
+      hostPlatform = crossSystem;
+      targetPlatform = crossSystem;
+
+      # Prior overrides are surely not valid as packages built with this run on
+      # a different platform, and so are disabled.
+      overrides = _: _: {};
+      extraBuildInputs = [ ]; # Old ones run on wrong platform
+      allowedRequisites = null;
+
+      cc = null;
+      hasCC = false;
+
+      extraNativeBuildInputs = old.extraNativeBuildInputs
+        ++ lib.optionals
+             (hostPlatform.isLinux && !buildPlatform.isLinux)
+             [ buildPackages.patchelf ]
```

Comment:
What's up with this? I know it's just being moved, but who's asking for patchelf on literally everything?

Reply:
No idea!

Reply:
I suppose [glass houses](https://github.com/rhelmot/nixpkgs/blob/a7cf88713e88a75074eaaf65c21692a2fc5542c1/pkgs/stdenv/freebsd/default.nix#L315) and so forth